src/System.Runtime.Extensions/tests/System/Environment.SetEnvironmentVariable.cs

```diff
@@ -31,9 +30,53 @@ public void IncorrectVariableThrowsArgument()
             AssertExtensions.Throws<ArgumentException>("variable", () => Environment.SetEnvironmentVariable(string.Empty, "test"));
             AssertExtensions.Throws<ArgumentException>("variable", () => Environment.SetEnvironmentVariable(NullString, "test"));
             AssertExtensions.Throws<ArgumentException>("variable", null, () => Environment.SetEnvironmentVariable("Variable=Something", "test"));
+        }
+
+        [Fact]
+        public void AllowAnyVariableLengths()
+        {
+            // longer than 32767
+            string longVar = new string('c', 40000);
+            string val = "test";
+
+            try
+            {
+                Environment.SetEnvironmentVariable(longVar, val);
+                Assert.Equal(val, Environment.GetEnvironmentVariable(longVar));
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(longVar, null);
+            }
+        }
+
+        [Fact]
+        public void AllowAnyVariableValueLengths()
+        {
+            string var = "Test_SetEnvironmentVariable_AllowAnyVariableValueLengths";
+            // longer than 32767
+            string longVal = new string('c', 40000);
+
+            try
+            {
+                Environment.SetEnvironmentVariable(var, longVal);
+                Assert.Equal(longVal, Environment.GetEnvironmentVariable(var));
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(var, null);
+            }
+        }
+
+        [Fact]
+        public void EnvironmentVariableTooLarge_Throws()
+        {
+            string var = "Test_SetEnvironmentVariable_EnvironmentVariableTooLarge_Throws";
```

Comment:
What about an excessively long environment variable name? We should have a test for it, whatever it does.